subsys/bluetooth/audio/bap_scan_delegator.c

```diff
@@ -743,6 +744,14 @@ static int scan_delegator_add_src(struct bt_conn *conn,
 			     subgroup->metadata_len);
 	}
 
+	if (scan_delegator_cp_cbs != NULL && scan_delegator_cp_cbs->add_source != NULL) {
+		err = scan_delegator_cp_cbs->add_source(conn, state);
+		if (err != BT_ATT_ERR_SUCCESS) {
+			LOG_DBG("add_source callback rejected: 0x%02x", err);
+			return err > 0 ? BT_GATT_ERR(err) : err;
```

Comment:
```suggestion
			/* Convert to a negative value if not already */
			return err > 0 ? BT_GATT_ERR(err) : err;
```

One alternative we can also consider is to assert on a negative value if we expect a BT_ATT_ERR value.

Reply:
As you told , Application tend to send -1 as error codes. Do we have to assert or can we convert as above ? what you think

Reply:
I think it's fair with an assert when we document it. 

The problem with a blind conversion is that `-1` would end up as `BT_ATT_ERR_INVALID_HANDLE`, which may/may not be a spec violation. I don't know if there are any specific rules w.r.t. sending ATT error responses. 

Your first iteration that just sent `BT_ATT_ERR_WRITE_REQ_REJECTED` made sense as well, as that is semantically what is going on. The only return values that makes sense for the application to send is `BT_ATT_ERR_WRITE_REQ_REJECTED`, `BT_ATT_ERR_PROCEDURE_IN_PROGRESS` and possibly `BT_ATT_ERR_UNLIKELY`, where `BT_ATT_ERR_WRITE_REQ_REJECTED` is likely the only one that will really get used. 

We could also force, via asserts or other methods, to only allow one of the 2/3 above, but it also seems like an unnecessary restriction.